announcement/announcement-tool/tool/src/java/org/sakaiproject/announcement/tool/AnnouncementAction.java

```diff
@@ -694,11 +701,42 @@ else if (view.equals(VIEW_MODE_PUBLIC))
 							{
 								messages = getMessagesPublic(site, channel, null, true, state, portlet);
 							}
+							else if (view.equals(VIEW_MODE_BYROLE))
+							{
+								messages = getMessagesByRoles(site, channel, null, true, state, portlet);
+							}
 						}
 						else
 						{
 							messages = getMessages(channel, null, true, state, portlet);
 						}
+						List<AnnouncementWrapper> finalMessages = new ArrayList<>();
+						for (AnnouncementWrapper message : messages){
+							site = SiteService.getSite(EntityManager.newReference(message.getReference()).getContext());
+							String selectedRoles = (String) message.getProperties().get("selectedRoles");
+							boolean addAnnounMessage = false;
+							if (selectedRoles == null) {
+								addAnnounMessage = true;
+							}else{
+								String[] selectedRolesArray = selectedRoles.replaceAll("[\\[\\]]", "").replaceAll(", ", ",").split(",");
+								Member actualMember = site.getMember(userDirectoryService.getCurrentUser().getId());
+								String memberRole = actualMember.getRole().getId();
+								for (String selectedRole : selectedRolesArray) {
+									if (selectedRole.equals(memberRole)) {
+										addAnnounMessage = true;
+									}
+								}
+								if (actualMember.getUserEid().equals(message.getAnnouncementHeader().getFrom().getDisplayId())){
+									addAnnounMessage = true;
+								} else if (actualMember.getUserId().equals("admin")) {
```

Comment:
You're hardcoding a user here, are you looking for securityService.isSuperUser() behavior?

Reply:
Yes, thanks, i will modify this and the others